commands/game.py

```diff
@@ -0,0 +1,935 @@
+        current_time = time.time() * 1000
+        current_event_id = ''
+        api = await self.fetch_api('https://bestdori.com/api/events/all.5.json')
+        for event in api:
+            if api[event]['startAt'][server]:
+                if float(api[event]['startAt'][server]) < current_time < float(api[event]['endAt'][server]):
+                    current_event_id = event
+                    break
+        if not current_event_id:
+            try:
+                for event in api:
+                    try:
+                        if current_time < float(api[event]['startAt'][server]):
+                            current_event_id = event
+                            break
+                    except TypeError:  # For between events
+                        continue
+            except KeyError:
+                current_event_id = list(api.keys())[-1]
+        if current_event_id:
+            return current_event_id
+        else:
+            return 0
+
+    async def get_event_time_left_sec(self, server: int, eventid: int):
+        current_time = time.time() * 1000.0
+        end_time = await self.get_event_end_time(server, eventid)
+        time_left_sec = (float(end_time) - current_time) / 1000
+        return time_left_sec
+
+    async def get_event_end_time(self, server: int, eventid: int):
+        api = await self.fetch_api(f'https://bestdori.com/api/events/{eventid}.json')
+        end_time = api['endAt'][server]
+        return float(end_time)
+
+    @game_commands.command(name='staruse',
+                           description='Provides star usage when aiming for a certain amount of EP.')
+    async def staruse(self,
+                      ctx: discord.ApplicationContext,
+                      ep_per_song: Option(int, "EP gained per song", required=True),
+                      current_ep: Option(int, "Current event point total", min_value=0, required=True),
+                      target_ep: Option(int, "Target event point total", min_value=0, required=True),
+                      flames_per_game: Option(int, "Flames used per game", choices=[1, 2, 3], required=True),
+                      current_player_rank: Option(int, "Current player rank", min_value=1, max_value=500, required=True),
+                      server: Option(str, "Server to get the current event's remaining time",
+                                     choices=[OptionChoice('EN', value='1'),
+                                              OptionChoice('JP', value='0'),
+                                              OptionChoice('TW', value='2'),
+                                              OptionChoice('CN', value='3'),
+                                              OptionChoice('KR', value='4')],
+                                     required=False,
+                                     default='1'),
+                      hrs_left: Option(float, "Manually input remaining time. Overrides time from 'server' argument.",
```

Comment:
leave this one as is because it's an advanced command and I can't figure out a way to make it more intuitive